pkg/alertmanager/operator.go

```diff
@@ -1063,7 +1063,7 @@ func (c *Operator) selectAlertmanagerConfigs(ctx context.Context, am *monitoring
 				"namespace", am.Namespace,
 				"alertmanager", am.Name,
 			)
-			c.eventRecorder.Eventf(amc, v1.EventTypeWarning, operator.InvalidConfigurationEvent, "AlertmanagerConfig %s was rejected due to invalid configuration: %v", amc.GetName(), err)
+			c.eventRecorder.Eventf(amc, nil, v1.EventTypeWarning, operator.InvalidConfigurationEvent, "AlertmanagerConfig %s was rejected due to invalid configuration: %v", amc.GetName(), err)
```

Comment:
Should we pass the workload resource (e.g. Alertmanager) as the related object?
```suggestion
			c.eventRecorder.Eventf(amc, am, v1.EventTypeWarning, operator.InvalidConfigurationEvent, "AlertmanagerConfig %s was rejected due to invalid configuration: %v", amc.GetName(), err)
```

Reply:
The [events.EventRecorder()](https://pkg.go.dev/k8s.io/client-go@v0.34.1/tools/internal/events#EventRecorder) interface has a different number of arguments compared to [tools.EventRecorder()](https://pkg.go.dev/k8s.io/client-go@v0.34.1/tools/internal/events#EventRecorder).

When I run the PR locally, the event's message contains a bit of garbage:

````
  message: 'foo/bar%!(EXTRA *fmt.wrapErrors=endpoints[0]: bearerTokenSecret: unable
    to get secret "foo": secrets "foo" not found)'
````

Reply:
Thanks @simonpasquier 
I miss it. 
Diff:
```
events.EventRecorder().Eventf(regarding runtime.Object, related runtime.Object, eventtype, reason, action, note string, args ...interface{})

recorder.EventRecorder().Eventf(object runtime.Object, eventtype, reason, messageFmt string, args ...interface{})
```
I miss `action`
Fixed